File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        
        List<Integer> list=new ArrayList<>();
        
        for(int i=0;i<arr.length;i++){
            // 리스트가 비어있으면 넣기
            if(list.isEmpty()){
                list.add(arr[i]);
            }
            // 리스트가 비어있지 않으면
            else{
                // 리스트의 마지막 값과 현재 숫자와 비교해서 다르면 리스트에 추가
                if(list.get(list.size()-1)!=arr[i]){
                    list.add(arr[i]);
                }
            }
        }

        // ArrayList를 array로 변환
        return list.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    # answer = []
    # for num in arr:
    #     if not answer or answer[-1] != num:  # 첫 원소거나 직전 값과 다르면 추가
    #         answer.append(num)
    # return answer
    
    answer=[arr[0]]
    # print(answer)
    for i in arr[1:]:
        if i != answer[-1]:
            answer.append(i)
    # print(answer)
    return answer